configs/myconfigs/sample_2d.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2016 Jason Lowe-Power
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Authors: Jason Lowe-Power, Brian Coutinho
+
+""" This script is an example of multi-threaded statistical sampling
+    where we simulate a sample multiple times to cover several possible
+    thread interleavings. Random pertubations are added during the
+    simulation to exercise mutliple execution paths/interleavings.
+
+    The sampling process thus works as two nested for-loops,
+    the outer one for each sample point in the program and
+    the inner one for mulitple observations at each sample point.
+
+    This script takes three arguments, the total number of instructions in the
+    ROI (which you can find by using runkvm.py), the number of sample
+    points you want to take during the execution, and the number of runs
+    to make at each sample point.
+
+    The output is a set of two levels of directories in the outdir.
+    At the top level is directory for each sample point which in turn
+    contains multiple sub-directories for each random observation.
+
+"""
+
+import os
+import sys
+import time
+
+import m5
+from m5.objects import *
+from m5.ticks import fromSeconds
+from m5.util.convert import toLatency
+
+sys.path.append('configs/common/') # For the next line...
+import SimpleOpts
+
+from system import MySystem
```

Comment:
Add a comment here that this is an example System object. You could replace this with the system you want to simulate.